src/shared/Microsoft.Git.CredentialManager.Tests/Interop/LibGit2Tests.cs

```diff
@@ -28,20 +30,97 @@ public void LibGit2_GetRepositoryPath_NotInsideRepository_ReturnsNull()
         [Fact]
         public void LibGit2_GetConfiguration_ReturnsConfiguration()
         {
-            var git = new LibGit2();
+            var trace = new NullTrace();
+            var git = new LibGit2(trace);
             using (var config = git.GetConfiguration())
             {
                 Assert.NotNull(config);
             }
         }
 
+        [Fact]
+        public void LibGit2Configuration_Enumerate_CallbackReturnsTrue_InvokesCallbackForEachEntry()
+        {
+            string repoPath = CreateRepository();
+            Git(repoPath, "config --local foo.name lancelot").AssertSuccess();
+            Git(repoPath, "config --local foo.quest seek-holy-grail").AssertSuccess();
+            Git(repoPath, "config --local foo.favcolor blue").AssertSuccess();
+
+            var expectedVisitedEntries = new List<(string name, string value)>
+            {
+                ("foo.name", "lancelot"),
+                ("foo.quest", "seek-holy-grail"),
+                ("foo.favcolor", "blue")
+            };
+
+            var trace = new NullTrace();
+            var git = new LibGit2(trace);
+            using (var config = git.GetConfiguration(repoPath))
+            {
+                var actualVisitedEntries = new List<(string name, string value)>();
+
+                bool cb(string name, string value)
+                {
+                    if (name.StartsWith("foo."))
+                    {
+                        actualVisitedEntries.Add((name, value));
+                    }
+
+                    // Continue enumeration
+                    return true;
+                }
+
+                config.Enumerate(cb);
```

Comment:
i'm guessing you are using "cb" to validate that we only have the 3 entries that start with 'foo' and they match.  May be worth a comment or two.

Reply:
> i'm guessing you are using "cb" to validate that we only have the 3 entries that start with 'foo' and they match.

Actually, no. The configuration object will contain more than just the three entries we explicitly add to the local configuration level in the test.

We filter to 'foo' section entries here to ignore any machine or user level entries which are 'noise' on the developer and build machines.

The test still checks that we enumerate entries in the configuration, and that we enumerate _at least_ those entries we expect.

The test below tests that enumeration can be stopped by the callback.